src/middlewared/middlewared/api/base/server/ws_handler/rpc.py

```diff
@@ -273,9 +273,34 @@ async def process_message(self, app: RpcWebSocketApp, message: Any):
         if method is None:
             app.send_error(id_, JSONRPCError.METHOD_NOT_FOUND.value, "Method does not exist")
             return
+        if not app.private_methods and method.private and not self._can_call_private_methods(app):
+            # FIXME: Eventually, prohibit this
+            self.middleware.logger.warning("Private method %r called on a connection without private_methods "
+                                           "enabled", method.name)
 
         asyncio.ensure_future(self.process_method_call(app, id_, method, message.get("params", [])))
 
+    def _can_call_private_methods(self, app: RpcWebSocketApp):
+        if app.origin.uid == 33:
+            # Proxied HexOS calls
```

Comment:
This comment is misleading. This will represent HexOS based connections but it will also represent nginx based connections. 33 is the `www-data` user and nginx forks workers and setsuid to that user and the `vendor_service.py` runs a systemd transient service as the `www-data` user for the HexOS based connection.